### PR TITLE
add a better error message for fetch when it fails with an unknown code

### PR DESCRIPTION
**Pull request by paperclover**
### What does this PR do?

Motivation: #9586 has a useless error message.

### CODE REVIEW

File: src/bun.js/webcore/response.zig [hunk 1]
```diff
@@ -1304,7 +1304,8 @@ pub const Fetch = struct {
                     error.STORE_LOOKUP => bun.String.static("Issuer certificate lookup error"),
                     error.NAME_CONSTRAINTS_WITHOUT_SANS => bun.String.static("Issuer has name constraints but leaf has no SANs"),
                     error.UNKKNOW_CERTIFICATE_VERIFICATION_ERROR => bun.String.static("unknown certificate verification error"),
-                    else => bun.String.static("fetch() failed. For more information, pass `verbose: true` in the second argument to fetch()"),
+
+                    else => |e| bun.String.createFormat("fetch() failed with {s}. For more information, pass `verbose: true` in the second argument to fetch()", .{@errorName(e)}) catch bun.outOfMemory(),
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore/response.zig]: maybe something like

```suggestion
                    else => |e| bun.String.createFormat("{s} fetching {}. For more information, pass `verbose: true` in the second argument to fetch()", .{@errorName(e), bun.fmt.quote(this.url) }) catch bun.outOfMemory(),
```

File: src/string.zig [hunk 2]
```diff
@@ -397,6 +397,14 @@ pub const String = extern struct {
         return BunString__fromUTF16(bytes.ptr, bytes.len);
     }
 
+    pub fn createFormat(comptime fmt: []const u8, args: anytype) !String {
+        var sba = std.heap.stackFallback(16384, bun.default_allocator);
```
Comment (jdalton) [inline src/string.zig:401]: Is `16384` a constant we can stash somewhere?

Comment (paperclover) [reply src/string.zig:401]: it is 2 ^ 14 bytes (16 kibibytes). this number is chosen arbitrarily.

File: src/bun.js/WebKit
Comment (gvilums) [inline src/bun.js/WebKit:1]: is the submodule update intentional?

**Review by gvilums — APPROVED** [review]

File: src/bun.js/WebKit
Comment (paperclover) [reply src/bun.js/WebKit:1]: no

**Pull request merged**